packages/contracts-bedrock/scripts/upgrades/PostSherlock.s.sol

```diff
@@ -0,0 +1,382 @@
+    string constant internal SystemConfig_Version = "1.2.0";
+    string constant internal L1ERC721Bridge_Version = "1.1.0";
+
+    /**
+     * @notice Place the contract addresses in storage so they can be used when building calldata.
+     */
+    function setUp() external {
+        implementations[GOERLI] = ContractSet({
+            L1CrossDomainMessenger: 0xfa37a4b2D49E21De63fa2b13D6dB213081E020b3,
+            L1StandardBridge: 0x79179704077E3324CC745A24a5CcC2a80A9B6842,
+            L2OutputOracle: 0x47bBB9054823f27B9B6A71F5cb0eBc785692FF2E,
+            OptimismMintableERC20Factory: 0xF516Fa87f89E4AC7C299aE28263e9EB851dE4781,
+            OptimismPortal: 0xa24A444C6ceeb1d4Fc19D1B78913C22B9d03BbC9,
+            SystemConfig: 0x2FFfe603caA9FA2C20E7F349138475a43284a6b1,
+            L1ERC721Bridge: 0xb460323429B08B9d1d427e6b8A450532988d5fe8
+        });
+
+        proxies[GOERLI] = ContractSet({
+            L1CrossDomainMessenger: 0x5086d1eEF304eb5284A0f6720f79403b4e9bE294,
+            L1StandardBridge: 0x636Af16bf2f682dD3109e60102b8E1A089FedAa8,
+            L2OutputOracle: 0xE6Dfba0953616Bacab0c9A8ecb3a9BBa77FC15c0,
+            OptimismMintableERC20Factory: 0x883dcF8B05364083D849D8bD226bC8Cb4c42F9C5,
+            OptimismPortal: 0x5b47E1A08Ea6d985D6649300584e6722Ec4B1383,
+            SystemConfig: 0xAe851f927Ee40dE99aaBb7461C00f9622ab91d60,
+            L1ERC721Bridge: 0x8DD330DdE8D9898d43b4dc840Da27A07dF91b3c9
+        });
+    }
+
+    /**
+     * @notice The entrypoint to this script.
+     */
+    function run(address _safe, address _proxyAdmin) external returns (bool) {
+        vm.startBroadcast();
+        bool success = _run(_safe, _proxyAdmin);
+        if (success) _postCheck();
+        return success;
+    }
+
+    /**
+     * @notice The implementation of the upgrade. Split into its own function
+     *         to allow for testability. This is subject to a race condition if
+     *         the nonce changes by a different transaction finalizing while not
+     *         all of the signers have used this script.
+     */
+    function _run(address _safe, address _proxyAdmin) public returns (bool) {
+        // Ensure that the required contracts exist
+        require(address(multicall).code.length > 0, "multicall3 not deployed");
+        require(_safe.code.length > 0, "no code at safe address");
+        require(_proxyAdmin.code.length > 0, "no code at proxy admin address");
+
+        GnosisSafe safe = GnosisSafe(payable(_safe));
+        uint256 nonce = safe.nonce();
+
+        bytes memory data = buildCalldata(_proxyAdmin);
+
+        // Compute the safe transaction hash
+        bytes32 hash = safe.getTransactionHash({
+            to: address(multicall),
+            value: 0,
+            data: data,
+            operation: Enum.Operation.DelegateCall,
+            safeTxGas: 0,
+            baseGas: 0,
+            gasPrice: 0,
+            gasToken: address(0),
+            refundReceiver: address(0),
```

Comment:
Is there a downside to specifying the `msg.sender` here rather than `address(0)`?

Reply:
I copied the arguments used in transactions that we have sent, `address(0)` generally represents default values and i believe it'll use `tx.origin` when its `address(0)`